src/main/groovy/life/qbic/utils/BioinformaticAnalysisParser.groovy

```diff
@@ -0,0 +1,369 @@
+     * @see {<a href="https://github.com/qbicsoftware/data-model-lib/blob/development/src/test/resources/examples/resultset/valid-resultset-example.json">valid datastructure example</a>
+     *
+     * After parsing all directories and files of the provided filestructure are contained in the children property of the root directory map entry.
+     * The underlying datastructure however expects a mapping of the expected folders and files to specific keys in the map outside of the children property such as
+     * QualityControl: { "name": "multiqc",
+     *                   "path": "./multiqc",
+     *                   "children": []
+     *                 }
+     * @param map a nested map representing the parsed fileTree structure
+     * @since 1.8.0
+     */
+    private static void adaptMapToDatasetStructure(Map map) {
+        List<Map> rootChildren = map.get("children") as List<Map>
+        List<Map> processFolders = []
+        rootChildren.each { currentChild ->
+            if (currentChild.containsKey("children")) {
+                //folder
+                String folderName = currentChild.get("name")
+                switch (folderName) {
+                    case "multiqc":
+                        insertAsProperty(map, currentChild, RequiredRootFolderKeys.QUALITYCONTROL.getKeyName())
+                        break
+                    case "pipeline_info":
+                        parsePipelineInformation(currentChild)
+                        insertAsProperty(map, currentChild, RequiredRootFolderKeys.PIPELINEINFORMATION.getKeyName())
+                        break
+                    default:
+                        processFolders.add(currentChild)
+                        break
+                }
+
+            } else if (currentChild.containsKey("file_type")) {
+                //file
+                switch (currentChild.get("name")) {
+                    case "run_id.txt":
+                        insertAsProperty(map, currentChild, RequiredRootFileKeys.RUNID.getKeyName())
+                        break
+                    case "sample_ids.txt":
+                        insertAsProperty(map, currentChild, RequiredRootFileKeys.SAMPLEID.getKeyName())
+                        break
+                    default:
+                        //ignore other files
+                        log.warn("Could not recognize file ${currentChild.path}")
+                        break
+                }
+
+            }
+        }
+        insertAsProperty(map, processFolders, RequiredRootFolderKeys.PROCESSFOLDERS.getKeyName())
+    }
+
+    /**
+     * Method which adapts the parsed content of the pipeline_info directory in place to the expected file structure.
+     * @see {<a href="https://github.com/qbicsoftware/data-model-lib/blob/development/src/test/resources/examples/resultset/valid-resultset-example.json">valid datastructure example</a>
```

Comment:
better reference master here, even though it is not yet there. Development branch is not stable.

Reply:
Good idea 👍